src/ethereum_test_tools/tests/test_code.py

```diff
@@ -308,3 +312,312 @@ def test_opcodes_if(conditional_bytecode: bytes, expected: bytes):
     Test that the if opcode macro is transformed into bytecode as expected.
     """
     assert bytes(conditional_bytecode) == expected
+
+
+@pytest.mark.parametrize(
+    "tx_data,switch_bytecode,expected_storage",
+    [
+        pytest.param(
+            to_hash_bytes(1),
+            Switch(
+                cases=[
+                    BytecodeCase(condition=Op.EQ(Op.CALLDATALOAD(0), 1), action=Op.SSTORE(0, 1)),
+                    BytecodeCase(condition=Op.EQ(Op.CALLDATALOAD(0), 2), action=Op.SSTORE(0, 2)),
+                ],
+                default_action=b"",
+            ),
+            {0: 1},
+            id="no-default-action-condition-met",
+        ),
+        pytest.param(
+            to_hash_bytes(0),
+            Switch(
+                cases=[
+                    BytecodeCase(condition=Op.EQ(Op.CALLDATALOAD(0), 1), action=Op.SSTORE(0, 1)),
```

Comment:
I think it would be awesome to have a function to generate the case with a calldata load as a condition, for simplicity.

Reply:
Great idea. I refactored your `case_calldata` helper function to a to a `CalldataCase` for a more consistent interface: 
https://github.com/danceratopz/execution-spec-tests/pull/25/commits/1473c07bd52c6a828808b2826dd3d8fc1942bd13#diff-fd9295703f4f6a3a2e215448c539eb882ded71beb144c7e2b9afebd2b1f4cbfbR260-R286

I didn't previously know about `field(init=False)`, but I think this seems like a valid and good use case.

I'll merge this PR as is, but plz speak up if you have other ideas, happy to add/modfy later.